modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/BlogPostingImageResourceImpl.java

```diff
@@ -173,6 +240,48 @@ private BlogPostingImage _toBlogPostingImage(FileEntry fileEntry)
 		};
 	}
 
+	private BlogPostingImage _updateBlogPostingImage(
+			FileEntry fileEntry, MultipartBody multipartBody)
+		throws Exception {
+
+		BinaryFile binaryFile = multipartBody.getBinaryFile("file");
+
+		if (binaryFile == null) {
+			throw new BadRequestException("No file found in body");
+		}
+
+		String title = null;
+		String viewableBy = null;
+
+		BlogPostingImage blogPostingImage =
+			multipartBody.getValueAsNullableInstance(
+				"blogPostingImage", BlogPostingImage.class);
+
+		if (blogPostingImage != null) {
+			title = blogPostingImage.getTitle();
+			viewableBy = blogPostingImage.getViewableByAsString();
+		}
+
+		if (title == null) {
+			title = fileEntry.getTitle();
+		}
+
+		if (viewableBy == null) {
+			viewableBy = Document.ViewableBy.OWNER.getValue();
+		}
+
+		return _toBlogPostingImage(
+			_dlAppService.updateFileEntry(
+				fileEntry.getFileEntryId(), binaryFile.getFileName(),
+				binaryFile.getContentType(), title, null, null, null,
+				DLVersionNumberIncrease.AUTOMATIC, binaryFile.getInputStream(),
+				binaryFile.getSize(), null, null, null,
+				ServiceContextBuilder.create(
+					fileEntry.getGroupId(), contextHttpServletRequest,
+					viewableBy
+				).build()));
```

Comment:
Blog images should be saved in the blogs private repository. We shouldn't be using the DLApp services here, but the PortletFileRepository instead.

Can you replace all usages of DLApp* by PortletFileEntry*? This may have been working out of chance.

Reply:
Hi @adolfopa, please check all commits again. 

* I've removed the `put` method, it is not useful for BlogPostingImages
* I'm using `PortletFileRepository` instead of the `DLApp` services
* I've added some tests